src/utils/Api.js

```diff
@@ -0,0 +1,65 @@
+class Api {
+    constructor(options) {
+      // constructor body
+      this.baseUrl = options.baseUrl
+      this.headers = options.headers
+    }
+
+    async makeRequest(endpoint, method = "GET", body = {}) {
+        let res = await fetch(this.baseUrl + endpoint, {
+            headers: this.headers,
+            method: method
+        });
+
+        if (method === "PATCH" || method === "POST") {
+            res = await fetch(this.baseUrl + endpoint, {
```

Comment:
* There are errors when submit any form https://snipboard.io/xfgCVZ.jpg 

Please, correct `makeRequest` like this, to make only **one** request (not 2)

```js

  async makeRequest(endpoint, method = "GET", body = {}) {
    let res;

    if (method === "PATCH" || method === "POST") {
      res = await fetch(this.baseUrl + endpoint, {
        method: method,
        headers: this.headers,
        body: JSON.stringify(body),
      });
    } else {
      res = await fetch(this.baseUrl + endpoint, {
        headers: this.headers,
        method: method,
      });
    }

    if (res.ok) {
      return res.json();
    }
    return await Promise.reject(`Error: ${res.status}`);
  }
```